### PR TITLE
Jump Game in Java

### LINKED ISSUE #3346: Jump_Game
## 🚀 Feature

  Jump Game (in Java) :
 
  We need to find the minimum number of jumps to reach the end of the array starting from the first element of the array,
  jumps value would be (-1) if we are not able to reach the end of the array
  
  
  example : array = {2,3,0,1,4,1,1,2,0,3,5} 
            minimum jumps = 5  {2 to 3, 3 to 4, 4 to 2, 2 to 3, 3 to 5}

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

Concept Building

## Pitch

Jump Game is one of the most asked interview question

### PR DESCRIPTION
**Pull request by Amanduttbhagat**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

We need to find the minimum number of jumps to reach the end of the array starting from the first element of the array,
jumps value would be (-1) if we are not able to reach the end of the array

example : array = {2,3,0,1,4,1,1,2,0,3,5}
minimum jumps = 5 {2 to 3, 3 to 4, 4 to 2, 2 to 3, 3 to 5}

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

fixed #3346

### CODE REVIEW

**Comment by Amanduttbhagat** [comment]
@HarshCasper @siddharth25pandey @iamrajiv please review my PR

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23 please review my PR

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Use proper code Format and align your code properly
- Remove the redundant comments
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code  
- Add a new line at the EOF

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23  can you please review it again
thank you

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23 please review the changes

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23  I have removed redundant comments and lines, if I'll remove more lines then it would be difficult to read

File: Java/cp/Jump_Game.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			int[] array = new int[number];
+
+			// Putting values in array using for loop
+			for(int i = 0 ; i < array.length ; i++) {
+				array[i] = sc.nextInt();
+			}
+			// Initializing the value returned by jumpGame method to jumps
+			int jumps = JumpGame.jumpGame(array);
+
+			System.out.println("minimum number of Jumps required : "+jumps);
```
Comment (ankitaggarwal23) [inline Java/cp/Jump_Game.java]: Remove two tabs space make it to one tab space.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rest looks good to merge.

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23 I have removed two tabs space and made it to one tab space.

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by Amanduttbhagat** [comment]
Thank you @ankitaggarwal23

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23 can you check #3592 also
Thank you

**Pull request merged**